added debug, fixed 'telimetry' to 'telemetry'

Timarrr opened the pull request.

Timarrr committed:
added debug, fixed 'telimetry' to 'telemetry'

Zauzolkov commented on Config.h:
```diff
@@ -6,12 +6,19 @@ namespace config {
-        constexpr int th_horizontal_back_right_pin  = 5;
-        constexpr int th_vertical_front_pin         = 6;
-        constexpr int th_vertical_back_pin          = 7;
+//        constexpr int th_horizontal_front_left_pin  = 2;
+//        constexpr int th_horizontal_front_right_pin = 3;
+//        constexpr int th_horizontal_back_left_pin   = 4;
+//        constexpr int th_horizontal_back_right_pin  = 5;
+//        constexpr int th_vertical_front_pin         = 6;
+//        constexpr int th_vertical_back_pin          = 7;
+
+        constexpr int th_horizontal_front_left_pin  = 6; //6
+        constexpr int th_horizontal_front_right_pin = 7; //7
+        constexpr int th_horizontal_back_left_pin   = 2; //2
+        constexpr int th_horizontal_back_right_pin  = 5; //5
+        constexpr int th_vertical_front_pin         = 4; //4
+        constexpr int th_vertical_back_pin          = 3; //3
```
You're about to push new config with your custom pinout. It's not okay to keep non-default pinout in master branch.

Zauzolkov commented on HighROV.cpp:
```diff
@@ -48,34 +52,96 @@ void debug(rov::RovControl &ctrl) {
+  if(SerialUSB.available()){
+    String msg = SerialUSB.readString();
+    msg.trim();
+    SerialUSB.print(msg=="reset" ? "Resetting the controller, it will explode in 3 seconds" : "");
```
Such jokes are fun but not okay.

Zauzolkov commented on Thrusters.h:
```diff
@@ -5,10 +5,9 @@ class Thrusters
 private:
     static Thrusters &inst();
-    int directions[6] = { 1,1,1,1,1,1 };
+    int directions[6] = { 1,-1,1,-1,-1,1 };
```
Are there any reasons to change default config?

Zauzolkov commented on HighROV.cpp:
```diff
@@ -48,34 +52,96 @@ void debug(rov::RovControl &ctrl) {
+        manip_pos += control.manipulatorOpenClose*10;
+        manip_pos = constrain(manip_pos, -100, 100);
+                
+        PWMController::set_thruster(7, manip_pos);
```
AFAIK, manipulator is controlled by PCA9536 chip, but not with Servo-PWM. Looks like you're trying to push changes, that reflects your custom (non-default) configuration of vehicle.

Zauzolkov commented on Config.h:
```diff
@@ -29,11 +36,17 @@ namespace config {
-        constexpr int manip_relese_ch   = 1;
+//        constexpr int manip_grab_ch     = 1;
+//        constexpr int manip_release_ch   = 0;
+//
+//        constexpr int manip_left_ch     = 2;
+//        constexpr int manip_right_ch    = 3;
+
+        constexpr int manip_grab_ch     = 2;
+        constexpr int manip_release_ch   = 3;
 
-        constexpr int manip_left_ch     = 2;
-        constexpr int manip_right_ch    = 3;
+        constexpr int manip_left_ch     = 0;
+        constexpr int manip_right_ch    = 1;
```
Custom (non-default) configs again.

Zauzolkov reviewed (changes requested):
Thank you for your contributions.

Anyway, I can't approve these changes right now. Please, don't try to push such changes, that will affect default configuration and pinout of HighRov vehicle.

Timarrr committed:
Further debug menu updates

Timarrr committed:
Debug menu and serial connection waiting fix

Timarrr committed:
Minor grammar and code fixes

Timarrr committed:
parent 90957fd61f5ddd986b8717446e1cfbc250e1ba21
author Timarrr <tigrmango@gmail.com> 1660660444 +1000
committer Timarrr <tigrmango@gmail.com> 1660741921 +1000

parent 90957fd61f5ddd986b8717446e1cfbc250e1ba21
author Timarrr <tigrmango@gmail.com> 1660660444 +1000
committer Timarrr <tigrmango@gmail.com> 1660741867 +1000

Reverted Config.h to pre-modified state

Removed MATE code and easter egg

Reverted Thrusters.h to default configuration

Remove redundant code

Motor init fix

Timarrr committed:
Reverted Config.h to pre-modified state

Removed MATE code and easter egg

Reverted Thrusters.h to default configuration

Reverting config and formatting to default

Remove redundant code

Tidying up code and fixing unfixed things

Motor init fix

fix code and possibly remove ip_decoder

Timarrr committed:
Reverted Config.h to pre-modified state

Timarrr committed:
Removed MATE code and easter egg

Timarrr committed:
Reverted Thrusters.h to default configuration

Timarrr committed:
Reverting config and formatting to default

Timarrr committed:
Remove redundant code

Timarrr committed:
Tidying up code and fixing unfixed things

Timarrr committed:
Motor init fix

Zauzolkov commented on HighROV.cpp:
```diff
@@ -12,80 +12,197 @@
 rov::RovControl control;
-rov::RovTelimetry telimetry;
+rov::RovTelemetry Telemetry;
+long long t_lm = 0;
```
What is `t_lm`? What does it mean?

Zauzolkov commented on HighROV.cpp:
```diff
@@ -12,80 +12,197 @@
+rov::RovTelemetry Telemetry;
+long long t_lm = 0;
+bool link_up = false;
+uint16_t debug_type = 0x00000000;
```
Max possible value for `uint16_t` is `0xFFFF`, so `0x0000` would be enough.

Zauzolkov commented on HighROV.cpp:
```diff
@@ -12,80 +12,197 @@
+	AnalogSensors::init();
 
-    delay(3000);
+	delay(250);
```
This delay for 3 seconds was intended for motors initialization, to prevent user control (and any other actions) while motors aren't ready.

Zauzolkov commented on HighROV.cpp:
```diff
@@ -12,80 +12,197 @@
+	if(debug_curr<0 || debug_curr>7){
+	SerialUSB.println("wrong input, please resend the input one number at a time");
+	}
+	debug_type = debug_type ^ (1 << (debug_curr - 1));
```
1) It will set `debug_type` variable, even if `wrong input`?
2) Use `bitSet` here?

Zauzolkov commented on HighROV.cpp:
```diff
@@ -12,80 +12,197 @@
+	}
+}
+
+void HighROV::run() {
+		using namespace config;
+		using namespace pwm;
+		serialHandler();
+		AnalogSensors::update();
+		if (Ethernet.linkStatus() == LinkON && !link_up) {
+			SerialUSB.println("Link status: On");
+			link_up = true;
+		}
+		else if (Ethernet.linkStatus() == LinkOFF && link_up) {
+			SerialUSB.println("Link status: Off. This usually indicates problems w/cable");
+			link_up = false;
+		}
```
What about moving Ethernet status checks to separate function in `Networking` or somewhere else?

In my opinion, main loop function (`run()`) became too verbose now.

Zauzolkov commented on HighROV.cpp:
```diff
@@ -12,80 +12,197 @@
+		} else {
+				analogWrite(LED_BUILTIN, 255);
+		}
+		if(bitRead(debug_type, 0)==1)//analog sensors debug
+		{
+			SerialUSB.println("[analog sensors debug]		Amperage:		" + String(AnalogSensors::getAmperage()) + "\n"\
+							 	 "                     			Voltage: 		" + String(AnalogSensors::getVoltage()));
+		}
+		if(bitRead(debug_type, 1)==1)//depth sensor debug
+		{
+			SerialUSB.println("[depth sensor debug] 			Depth:			" + String(AnalogSensors::getAmperage()) + "\n");
+		}
+		if(bitRead(debug_type, 2)==1)//IMU debug
+		{
+			SerialUSB.println("[IMU debug] 					Yaw/Roll/Pitch:	" + String(IMUSensor::getYaw(), 2) + "/" + String(IMUSensor::getRoll(),2) + "/" + String(IMUSensor::getPitch()));
+		}
+		if(bitRead(debug_type, 3)==1)//Manipulator debug
+		{
+			SerialUSB.println("[Manipulator debug] 			Position:		" + String(control.manipulatorOpenClose) + "\n"\
+								 "								Rotation:		" + String(control.manipulatorRotation));
+		}
+		if(bitRead(debug_type, 4)==1)//Networking debug
+		{
+			SerialUSB.println("[Networking debug]			" + Networking::status());
+		}
+		if(bitRead(debug_type, 5)==1)//Networking debug
+		{
+			SerialUSB.println("[Thrusters debug]				" + Thrusters::status);
+		}
+		if(bitRead(debug_type, 6)==1)//Reset debug
+		{
+			debug_type = 0x00000000;
+		}
```
1) What about moving this to another function? (or maybe new class related to your new debug features) Looks like there are too much things in `run()` now, that can be replaced with small functions calls.
2) You can use tabulations `\t` instead of spaces when formatting serial output. These bunches of spaces are quite "ugly"; lines of code are too long; overall code style looks somewhat messy.
3) Probably it would be better to use `enum` instead of **unnamed** constants (`debug_type` values).

Zauzolkov commented on HighROV.cpp:
```diff
@@ -12,80 +12,197 @@
+		if(msg=="reset"){
+			SerialUSB.print("Resetting the controller, please reconnect the debug cable or serial monitor if you want to continue debugging");
+			gracefulReset();
+		}
+		else
+		if(msg=="debug")
+			debugMenu();
+		else
+		SerialUSB.println("Send \"reset\" for controller reset or \"debug\" for debug menu");
+	}
```
Weird code formatting.

Zauzolkov commented on HighROV.cpp:
```diff
@@ -12,80 +12,197 @@
+			link_up = false;
+		}
+		if ((link_up && (millis() >= t_lm + 500)) || millis() >= t_lm + 5000){//if link is up -> check every .5 of a second, otherwise check once every 5 seconds
+			byte status = Ethernet.maintain();
```
> *`Ethernet.maintain()` – Allows for the renewal of DHCP leases.*

AFAIK, IP address of board is static. Is it tested with default router configuration?
Do we need to handle DHCP if the default IP config is static?

Zauzolkov commented on Networking.cpp:
```diff
@@ -1,74 +1,106 @@
+    using namespace config::networking;
+
+    return "\nTarget IP:        " + String(selfip[0]) + "." + String(selfip[1]) + "." + String(selfip[2]) + "." + String(selfip[3]) + "\n"\
+           "Current IP:         " + ip_decoder(Ethernet.localIP()) + "\n"\
+           "DNS server IP:      " + ip_decoder(Ethernet.dnsServerIP()) + "\n"\
+           "Gateway IP:         " + ip_decoder(Ethernet.gatewayIP()) + "\n"\
+           "Subnet mask:        " + ip_decoder(Ethernet.subnetMask()) + "\n"\
+           "Link status:        " + (Ethernet.linkStatus() ? "ON\n" : "OFF\n");
```
Did you tried `Serial.print(Ethernet.localIP());`?

Zauzolkov commented on Networking.cpp:
```diff
@@ -1,74 +1,106 @@
+    Udp.endPacket();
+}
+
+String Networking::ip_decoder(uint32_t addr){
+    byte ip_buf[4]; // 111.222.333.444
+	for(int i = 3; i >= 0; i--){ //byte segment
+		for(int j = 7; j >=0; j--){ //bit
+			ip_buf[i] += bitRead(addr, i*8+j) * pow(2,j); //bit read order - LSB first -> reading MSB first
+		}
+	}
+    return String(ip_buf[0]) + "." + String(ip_buf[1]) + "." + String(ip_buf[2]) + "." + String(ip_buf[3]) + "." ;
+}
```
It can be done in much easier and less verbose way. But I doubt that we need it here at all…

Zauzolkov commented on Thrusters.cpp:
```diff
@@ -1,42 +1,48 @@
+		PWMController::set_thruster(vertical_front, constrain(data.directions[4] * z, -100, 100));
+		PWMController::set_thruster(vertical_back,  constrain(data.directions[5] * z, -100, 100));
+
+		status = ("Axis X:	" + (String) x + "	Axis Y:	" + (String) y + "	Axis W:	" + (String) w + "	Axis Z:	" + z + "\n"\
+							"Thruster HorFrontLeft:	" + (String) (data.directions[0] * (y + x + w)) + "	Thruster HorFrontRight	" + (String) (data.directions[1] * (y - x - w)) +\
+					 "\n" + "Thruster HorBackLeft:	" + (String) (data.directions[2] * (y - x + w)) + "	Thruster HorBackRight	" + (String) (data.directions[3] * (y + x - w)) +\
+					 "\n" + "Thruster VerFront :	" + (String) (data.directions[4] * z) +           "	Thruster VerBack	" + (String) (data.directions[5] * z));
```
1) What about splitting these long code lines to smaller ones?
2) You are performing same calculations twice.

Timarrr committed:
fix code and possibly remove ip_decoder

Timarrr committed:
Merge branch 'master' of https://github.com/Timarrr/HighROV_firmware

Timarrr committed:
Merge branch 'master' of https://github.com/Timarrr/HighROV_firmware

Timarrr committed:
Revert "fix code and possibly remove ip_decoder"

This reverts commit de7494b373e9b4b5987ebdafcac4877a96dffe7f.

Timarrr committed:
Update README.md
